neo4j/__init__.py

```diff
@@ -1097,10 +1101,12 @@ def __init__(self, **metadata):
         self.parameters = metadata.get("parameters")
         self.statement_type = metadata.get("type")
         self.counters = SummaryCounters(metadata.get("stats", {}))
-        self.result_available_after = metadata.get("result_available_after")
-        self.result_consumed_after = metadata.get("result_consumed_after")
-        self.t_first = metadata.get("t_first")
-        self.t_last = metadata.get("t_last")
+        if self.protocol_version < BOLT_VERSION_3:
```

Comment:
I would prefer protocol versions not leaked here. The best would the protocol specific implementation goes to protocol classes. If it too much work, then I would prefer:

```
self.result_available_after = metadata.get("t_first") == null ? metadata.get("result_available_after") : metadata.get("t_first")
self.result_consumed_after = metadata.get("t_last") == null ? metadata.get("result_consumed_after") : metadata.get("t_first")
```